### ISSUE TITLE
Sharing from some apps and returning opens the shared link in the original app

### ISSUE THREAD

**Issue by kiikoh** [achorein/expo-share-intent#103]
**Describe the bug**
Sharing from spotify (maybe others) and returning opens the shared link in the original app. You don't have to directly return to the app, you can go to home screen first, or open other app in between. I don't know why spotify has this issue specifically, but I assume others might also

The video linked is a recording from an iPhone SE gen1 on ios 15.8, but I have confirmed the same issue occurs on my friends modern devices also. I think a relevant detail is the quick flash of the sharing dialog before the testing app opens. That dialog doesn't show on modern devices though, only as a white flash. 

ignore that the app doesn't open to a different page. That was done intentionally to reduce surface area.
https://github.com/user-attachments/assets/c2b20074-c3a2-4817-8b5c-0285a3500436

**To Reproduce**
I have linked a fresh install of expo router tabs with expo-share-intent.
https://github.com/kiikoh/spotify-share-bug-repro

**Environment**
```
System:
  OS: Linux 5.15 Ubuntu 20.04.6 LTS (Focal Fossa)
  CPU: (12) x64 AMD Ryzen 5 5600X 6-Core Processor
  Memory: 8.78 GB / 9.72 GB
  Shell:
    version: "5.8"
    path: /usr/bin/zsh
Binaries:
  Node:
    version: 20.11.1
    path: /usr/local/bin/node
  Yarn:
    version: 1.22.11
    path: /usr/local/bin/yarn
  npm:
    version: 10.2.4
    path: /usr/local/bin/npm
  Watchman: Not Found
SDKs:
  Android SDK: Not Found
IDEs:
  Android Studio: Not Found
Languages:
  Java:
    version: 11.0.19
    path: /usr/bin/javac
  Ruby: Not Found
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.74.5
    wanted: 0.74.5
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: Not found
  newArchEnabled: Not found
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found

info React Native v0.75.3 is now available (your project is running on v0.74.5).
info Changelog: https://github.com/facebook/react-native/releases/tag/v0.75.3
info Diff: https://react-native-community.github.io/upgrade-helper/?from=0.74.5
info For more info, check out "https://reactnative.dev/docs/upgrading?os=linux".
```
**Additional context**

- expo version (ex: 49) : 51
- using firebase : no
- using static build : dont know what this mean... I built with eas
- routing : expo-router
- package version (ex: 1.2.0) : 2.3.0
- platform target : iOS
- device : real device
- type: expo build

**Comment by kiikoh** [comment]
This bug appears to have been fixed somewhere between 2.3.0 and 2.5.2. Thanks!

**Comment by achorein** [comment]
🙌  
v2.4.0 has resolved a threading issue in the ios share extension, sounds great !
